ApiIndex/ServiceModel.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf.Reflection;
+using System.Linq;
+using static Google.Cloud.Tools.ApiIndex.V1.Method.Types;
+
+namespace Google.Cloud.Tools.ApiIndex
+{
+    /// <summary>
+    /// Internal representation of a service, without reference to any specific schema version.
+    /// </summary>
+    public class ServiceModel
+    {
+        private ServiceDescriptor service;
+
+        public ServiceModel(ServiceDescriptor service)
+        {
+            this.service = service;
+        }
+
+        public V1.Service ToV1Service() => new V1.Service
+        {
+            FullName = service.FullName,
+            ShortName = service.Name,
+            Methods = { service.Methods.OrderBy(method => method.Name).Select(ConvertMethod) }
```

Comment:
As discussed by email, it would be nice to add here the google.api.http (https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L29) message included in the custom options field of the rpc.

For our use-case a simplified JSON representation could be a flat list of repeated HttpRule messages with only the "oneof pattern" and body. e.g:

https://github.com/googleapis/googleapis/blob/master/google/cloud/iot/v1/device_manager.proto#L107

```
...
"httpRules": [{
  "patch": "/v1/{device.name=projects/*/locations/*/registries/*/devices/*}",
  "body": "device"
},{
  "patch": "/v1/{device.name=projects/*/locations/*/registries/*/groups/*/devices/*}",
  "body": "device"
}]
...
```

Reply:
Having discussed this internally, we do indeed need something where different bindings can have different HTTP methods. We don't need to represent the body, however.

I'll create another commit that uses a representation of:

```json
"httpRules": [{
  "method": "patch",
  "path": "/v1/{device.name=projects/*/locations/*/registries/*/devices/*}"
},{
  "method": "patch",
  "path": "/v1/{device.name=projects/*/locations/*/registries/*/groups/*/devices/*}"
}]
```

(I'd expect that having the method as the value of a field rather than as a key in the object should be easier to consume.)